File: Python3/Medium/RuntimeError/plates-between-candles_8.py
class Solution:
    def platesBetweenCandles(self, s: str, queries: List[List[int]]) -> List[int]:
        
        # The idea is to use prefix sums to find the number of plates between candles in a given substring.
        # The number of plates between candles in a substring is the number of plates in the substring minus the number of plates in the left and right substrings.
        # The number of plates in a substring is the number of plates in the prefix sum array for the substring.
        # The number of plates in the left substring is the number of plates in the prefix sum array for the left substring.
        # The number of plates in the right substring is the number of plates in the prefix sum array for the right substring.
        # The number of plates in the left substring is the number of plates in the prefix sum array for the left substring.
        # The number of plates in the right substring is the number of plates in the prefix sum array for the right substring.
        # The number of plates in the left substring is the number of plates in the prefix sum array for the left substring.
        # The number of plates in the right substring is the number of plates in the prefix sum array for the right substring.
        # The number of plates in the left substring is the number of plates in the prefix sum array for the left substring.
        # The number of plates in the right substring is

File: Java/Medium/WrongAnswer/plates-between-candles_8.java
class Solution {
    public int[] platesBetweenCandles(String s, int[][] queries) {
        
        int[] candles = new int[s.length()];
        int[] plates = new int[s.length()];
        int[] ans = new int[queries.length];
        int count = 0;
        for(int i = 0; i < s.length(); i++){
            if(s.charAt(i) == '|'){
                count++;
            }
            candles[i] = count;
        }
        count = 0;
        for(int i = s.length()-1; i >= 0; i--){
            if(s.charAt(i) == '|'){
                count++;
            }
            plates[i] = count;
        }
        for(int i = 0; i < queries.length; i++){
            int left = queries[i][0];
            int right = queries[i][1];
            int leftCandle = candles[left];
            int rightCandle = plates[right];
            ans[i] = Math.max(0, leftCandle + rightCandle - 2);
        }
        return ans;
    }
}